### PR TITLE
Adds lint/rules/jsx-a11y/noOnChange

### PR DESCRIPTION
**Pull request by smlmrkhlms**
Reference Issue: #341 

This is my first contribution to the project so let me know if any of my additions are not in line with this project's existing patterns. This pull request adds support for the [jsx-a11y/no-onchange](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-onchange.md) lint rule.

### CODE REVIEW

File: packages/@romejs/diagnostics/categories.ts [hunk 1]
```diff
@@ -153,6 +153,7 @@ type LintDiagnosticCategory =
 	| "lint/jsx-a11y/noAccessKey"
 	| "lint/jsx-a11y/noAutofocus"
 	| "lint/jsx-a11y/noDistractingElements"
+	| "lint/jsx-a11y/noOnchange"
```
Comment (bitpshr) [inline packages/@romejs/diagnostics/categories.ts:156]: Nit: Could you fix the casing of this rule's display name (and its files) to be `noOnChange`? Also, just an FYI for future reference, there's a lint rule generation script at `scripts/lint/add.cjs` that handles most of these updates (and the name casing) automatically:

```
node scripts/lint/add.cjs no-on-change jsx-a11y
```

File: packages/@romejs/diagnostics/descriptions.ts [hunk 2]
```diff
@@ -352,6 +352,10 @@ export const descriptions = createMessages({
 			category: "lint/jsx-a11y/noDistractingElements",
 			message: `Do not use ${element} elements as they can create visual accessibility issues and are deprecated.`,
 		}),
+		JSX_A11Y_NO_ONCHANGE: {
+			category: "lint/jsx-a11y/noOnchange",
+			message: `<emphasis>onBlur</emphasis> should be used in favor of <emphasis>onChange</emphasis>. Only use <emphasis>onChange</emphasis> if absolutely necessary without negatively affecting keyboard only or screen reader users.`,
```
Comment (bitpshr) [inline packages/@romejs/diagnostics/descriptions.ts:357]:
```suggestion
			message: "<emphasis>onBlur</emphasis> should be used in favor of <emphasis>onChange</emphasis>. Only use <emphasis>onChange</emphasis> if absolutely necessary without negatively affecting keyboard only or screen reader users.",
```

**Comment by smlmrkhlms** [comment]
@bitpshr Thank you for the initial review and letting me know about the lint addition script. I will for sure use it in the future. I updated the PR with your feedback. Thanks again!

**Comment by bitpshr** [comment]
Thanks for this @smlmrkhlms. I believe CI is failing because the filename on disk is still incorrectly cased, e.g. `jsx-a11y/noOnchange.ts` instead of `jsx-a11y/noOnChange.ts`. Everything should pass once that's fixed.

**Comment by smlmrkhlms** [comment]
@bitpshr Thanks for you help!

**Comment by sebmck** [comment]
Looks good! Thanks for the contribution @smlmrkhlms! If you need any help in the future feel free to join our [Discord](https://discord.gg/rome). Thank you!

**Comment by smlmrkhlms** [comment]
@sebmck Thank you! I will bring my questions there in the future. I am going to do my best to continue contributing whenever I can. This project is incredibly exciting. Thank you for open-sourcing it.

**Pull request merged**